### ISSUE TITLE
[Meta] Forge Class & Method Renames for 1.19

### ISSUE THREAD

**Issue by TheCurle** [MinecraftForge/MinecraftForge#8165]
==== Why =====

A few people have requested that we rename some Forge classes and methods; whether to align more properly with the Mojang Mappings, or because they have substantially changed from their first design.

This issue will be used to collect potential renames, and we'll decide which ones to take action on when the time for the update comes.

==== How ====

If you want to suggest a rename, just leave it in a comment here, WITH a reason.
Suggestions with no context will just be ignored, as is the sensible option.

**Comment by alcatrazEscapee** [comment]
`MobSpawnInfoBuilder` -> `MobSpawnSettingsBuilder`
Reason: it's a builder that was directly named after the (mcp) class name at the time. That name has now changed.

https://github.com/MinecraftForge/MinecraftForge/blob/9e476551be4649c35881c103748f3bdc3ca3b353/src/main/java/net/minecraftforge/common/world/MobSpawnInfoBuilder.java#L32

`TileFluidHandler` -> `FluidHandlerBlockEntity` (Alternatively, just delete this, is anyone actually using this? Why can't this be a wiki page with some "here's what you need to implement a fluid handler tile entity?)
Reasons:
- It's named from pre 1.14 convention (prefix vs. suffix)
- Tile entity has become block entity.

https://github.com/MinecraftForge/MinecraftForge/blob/9e476551be4649c35881c103748f3bdc3ca3b353/src/main/java/net/minecraftforge/fluids/capability/TileFluidHandler.java#L36

**Comment by TheOnlyTails** [comment]
### Remove the leading `I` from the names of interfaces

This no longer fits in with the vanilla classes,  since there are no longer any interfaces with a leading `I` (AFAIK).

**Comment by ChampionAsh5357** [comment]
### Renaming Extension Interfaces

Since most of our extension interfaces rely on the concept of `Forge*`, any abstraction also has a similar name (e.g. `ForgeFlowingFluid` would need to be renamed if the leading `I` was removed for an interface). As such, all extension interfaces should be suffixed with the word `Extension` (e.g. `IForgeBlock` -> `BlockExtension`/`ForgeBlockExtension`) for a larger clarity on what they are and what Forge is applying them for.

### Flip Prefixed Type Names

Most of the types in Forge still use the old pre-1.14 syntax of naming where the type is prefixed to the name (e.g. `SlotItemHandler`). These should now be flipped to reflect the mojang style of naming: `ItemHandlerSlot`.

**Comment by PepperCode1** [comment]
`ItemModelMesherForge` -> `ForgeItemModelShapes`
- Follow Mojmap: this class extends `ItemModelShapes`

`ModelLoader` -> `ForgeModelBakery`
- Prevent confusion with `IModelLoader`, which is in the same package and is completely unrelated to this class
- Follow Mojmap: this class extends `ModelBakery`

**Comment by LexManos** [comment]
> ### Remove the leading `I` from the names of interfaces
> This no longer fits in with the vanilla classes, since there are no longer any interfaces with a leading `I` (AFAIK).

Vetoed. Vanilla is vanilla, Forge is Forge. We can have our own style. And this is one I like so we're not droping. So far the rest I don't really care about.

**Comment by Alpvax** [comment]
> Vetoed. Vanilla is vanilla, Forge is Forge. We can have our own style

Regarding the people unhappy with this, remember that there was a style vote during the large re-write of forge, and the majority of people wanted to keep this.

As to the renaming of classes to match their vanilla name, I think that is generally a good idea.

**Comment by TheCurle** [comment]
By "i don't really care about", he means that he has no strong opinions either way, and we plan to continue with that idea.

**Comment by Alpvax** [comment]
> By "i don't really care about", he means that he has no strong opinions either way, and we plan to continue with that idea.

I realise that, I didn't mean it to sound like I was disagreeing with Lex, just that as I was already commenting, I would throw my support behind the renaming.

**Comment by sciwhiz12** [comment]
Closing as 1.18 has now released with these renames performed. If there are any names which you feel need to also be renamed, please raise this issue in the official Discord. Thank you.

**Comment by PepperCode1** [comment]
I was asked to put the new renames I thought of here, that would be done for 1.18.1.

- `GatherDataEvent` -> `event` or `data.event` package
Reason: this event is currently in the old `forge.event.lifecycle` package by itself.
- `client.ItemModelMesherForge` -> `client.model.ForgeItemModelShapes`
Reason: `ItemModelShapes` is the Mojmap name for `ItemModelMesher`.
- `BlockEntity.getTileData` -> `getPersistentData`
Reason: remove use of `tile` and make consistent with `Entity.getPersistentData`.

**Comment by TheCurle** [comment]
From here on, renames will be targetted for 1.19.
The same process applies.

**Comment by alcatrazEscapee** [comment]
> `TileFluidHandler` -> `FluidHandlerBlockEntity` (Alternatively, just delete this, is anyone actually using this? Why can't this be a wiki page with some "here's what you need to implement a fluid handler tile entity?) Reasons:
> 
> * It's named from pre 1.14 convention (prefix vs. suffix)
> * Tile entity has become block entity.
> 
> https://github.com/MinecraftForge/MinecraftForge/blob/63a42bad984f9a9fe2fa8ce92e95b0e0c9064d04/src/main/java/net/minecraftforge/fluids/capability/TileFluidHandler.java#L36

Same request as above, but referencing the class in 1.18 now.

**Comment by Random832** [comment]
`NBTIngredient` ->`NbtIngredient`, every Mojang class with nbt in it spells it this way

**Comment by josephtarbit** [comment]
> `NBTIngredient` ->`NbtIngredient`, every Mojang class with nbt in it spells it this way

On a related note: `INBTSerializable` -> `INbtSerializable` and in the same class `serializeNBT` -> `serializeNbt` and `deserializeNBT` -> `deserializeNbt`

**Comment by PepperCode1** [comment]
Rename the following methods in `IForgeItem` and `IForgeItemStack` to the corresponding Mojmap names of the equivalent vanilla methods:
- getContainerItem
- hasContainerItem
- getCreativeTabs
- getItemStackLimit
- getItemEnchantability
- isDamageable

**Comment by SizableShrimp** [comment]
Some ForgeRegistries registries and keys are inconsistent relative to the classname / Key, so I think all of these changes are reasonable:
* `ForgeRegistries.Keys.CONTAINER_TYPES` -> `ForgeRegistries.Keys.MENU_TYPES`
* `ForgeRegistries.CONTAINERS` -> `ForgeRegistries.MENU_TYPES`
* `ForgeRegistries.ENTITIES` -> `ForgeRegistries.ENTITY_TYPES`
* `ForgeRegistries.BLOCK_ENTITIES` -> `ForgeRegistries.BLOCK_ENTITY_TYPES`
* `ForgeRegistries.PROFESSIONS` -> `ForgeRegistries.VILLAGER_PROFESSIONS`

`ForgeRegistries.PAINTING_TYPES` is weird because the Mojang classname is "Motive", not sure if we should make that name worse.

`ForgeRegistries.DATA_SERIALIZERS` is quite generic, I think it would be best to rename it to start with `ENTITY_`.
Same thing for `ForgeRegistries.LOOT_MODIFIER_SERIALIZERS`, should this be prefixed with `GLOBAL_` to match the classname?

**Comment by eerussianguy** [comment]
`PotionEvent` -> `MobEffectEvent` and it's children

These events happen when a mob effect is applied to an entity, not necessarily a potion. I think this is a holdover from the old name.

**Comment by XFactHD** [comment]
- `WorldEvent` -> `LevelEvent`
- `RenderBlockOverlayEvent` -> `RenderScreenEffectEvent` (Mojang calls the overlays when swimming in fluids "screen effects")
- `EntityJoinWorldEvent` -> `EntityJoinLevelEvent`
- `EntityLeaveWorldEvent` -> `EntityLeaveLevelEvent`

Not sure on these two as the referenced classes still have "Container" in their name (`AbstractContainerMenu` and `AbstractContainerScreen` respectively)
- `PlayerContainerEvent` -> `PlayerMenuEvent`
- `ContainerScreenEvent` -> `MenuScreenEvent`

**Comment by Luis-St** [comment]
renaming of classes:
`TickEvent.WorldTickEvent` -> `TickEvent.LevelTickEvent`
`TickEvent.Type.WORLD` -> `TickEvent.Type.LEVEL`

renaming getters and thier field names of Events:
`EnchantmentLevelSetEvent#getWorld` -> `EnchantmentLevelSetEvent#getLevel` (update to Mojang mappings)
`ItemEvent#getEntityItem` -> `ItemEvent#getItemEntity` (update to Mojang mappings)
`LivingEvent#getEntityLiving` -> `LivingEvent#getLiving` (update to Mojang mappings)
`LivingSpawnEvent#getWorld` -> `LivingSpawnEvent#getLevel` (update to Mojang mappings)
`AnvilRepairEvent#getItemResult` -> `AnvilRepairEvent#getOutput` (to match with getter of AnvilUpdateEvent)
`AnvilRepairEvent#getItemInput` -> `AnvilRepairEvent#getLeft` (to match with getter of AnvilUpdateEvent)
`AnvilRepairEvent#getIngredientInput` -> `AnvilRepairEvent#getRight` (to match with getter of AnvilUpdateEvent)
`ArrowLooseEvent#getWorld` -> `ArrowLooseEvent#getLevel` (update to Mojang mappings)
`ArrowNockEvent#getWorld` -> `ArrowNockEvent#getLevel` (update to Mojang mappings)
`BonemealEvent#getWorld` -> `BonemealEvent#getLevel` (update to Mojang mappings)
`FillBucketEvent#getWorld` -> `FillBucketEvent#getLevel` (update to Mojang mappings)
`PlayerSetSpawnEvent#getSpawnWorld` -> `PlayerSetSpawnEvent#getSpawnLevel` (update to Mojang mappings)
`PlayerWakeUpEvent#updateWorld` -> `PlayerWakeUpEvent#updateLevel` (update to Mojang mappings)
`EntityJoinWorldEvent#getWorld` -> `EntityJoinWorldEvent#getLevel` (update to Mojang mappings)
`EntityLeaveWorldEvent#getWorld` -> `EntityLeaveWorldEvent#getLevel` (update to Mojang mappings)
`EntityMountEvent#getWorldObj` -> `EntityMountEvent#getLevel` (update to Mojang mappings)
`VillageSiegeEvent#getWorld` -> `VillageSiegeEvent#getLevel` (update to Mojang mappings)
`BlockEvent#getWorld` -> `BlockEvent#getLevel` (update to Mojang mappings)
`BlockEvent.CreateFluidSourceEvent#getWorld` -> `BlockEvent.CreateFluidSourceEvent#getLevel` (update to Mojang mappings)
`ChunkWatchEvent#getWorld` -> `ChunkWatchEvent#getLevel` (update to Mojang mappings)
`ExplosionEvent#getWorld` -> `ExplosionEvent#getLevel` (update to Mojang mappings)
`WorldEvent#getWorld` -> `WorldEvent#getLevel` (update to Mojang mappings)
`FOVModifierEvent#getEntity` -> `FOVModifierEvent#getPlayer` (to match with return type of getter)

**Comment by SizableShrimp** [comment]
`WorldCapabilityData` -> `LevelCapabilityData`

**Comment by KnightMiner** [comment]
`NBTIngredient` -> `StrictNBTIngedient`. #8517 added `PartialNBTIngredient`, so a rename would help differentiate it. Also change the ingredients serializer ID to `forge:strict_nbt`

**Comment by IchHabeHunger54** [comment]
`LivingUpdateEvent` -> `LivingTickEvent` (for consistency with `PlayerTickEvent`)

**Comment by sciwhiz12** [comment]
> `ForgeRegistries.DATA_SERIALIZERS` is quite generic, I think it would be best to rename it to start with `ENTITY_`.
> Same thing for `ForgeRegistries.LOOT_MODIFIER_SERIALIZERS`, should this be prefixed with `GLOBAL_` to match the classname?

@SizableShrimp: Would that entail changing the keys of the registries to `entity_data_serializers` and `forge:global_loot_modifier_serializers` as well?

> Not sure on these two as the referenced classes still have "Container" in their name (`AbstractContainerMenu` and `AbstractContainerScreen` respectively)
>   * `PlayerContainerEvent` -> `PlayerMenuEvent`
>   * `ContainerScreenEvent` -> `MenuScreenEvent`

@XFactHD: I believe `ContainerScreenEvent` should remain as it is, to correspond with the `AbstractContainerScreen` class it is created for. 

However, I am split on `PlayerContainerEvent`. On one hand, the class it holds, `AbstractContainerMenu`, still has "container" in the name, still making it a valid name. On the other hand, it is also good to communicate that this refers to the container menu and not `Container` (the Mojang name for MCP's `IInventory`).

Perhaps that should be more publicly discussed for opinions, but until a decision to rename is made and agreed to, I think the name should be kept as-is to keep the status quo.

**Comment by sciwhiz12** [comment]
> `ItemEvent#getEntityItem` -> `ItemEvent#getItemEntity` (update to Mojang mappings)
> `LivingEvent#getEntityLiving` -> `LivingEvent#getLiving` (update to Mojang mappings)

@Luis-st: Instead of renaming the existing getters, it may perhaps be better to override `#getEntity` and narrow down the return type instead to e.g. `ItemEntity` and `LivingEntity` (and perhaps `PlayerEvent`) using the covariance of method return types. What do you think?

**Comment by IchHabeHunger54** [comment]
> However, I am split on `PlayerContainerEvent`. On one hand, the class it holds, `AbstractContainerMenu`, still has "container" in the name, still making it a valid name. On the other hand, it is also good to communicate that this refers to the container menu and not `Container` (the Mojang name for MCP's `IInventory`).

what about `PlayerContainerMenuEvent`? or is that too long of a name?

**Comment by ghost** [comment]
> Some ForgeRegistries registries and keys are inconsistent relative to the classname / Key, so I think all of these changes are reasonable:
> 
> * `ForgeRegistries.Keys.CONTAINER_TYPES` -> `ForgeRegistries.Keys.MENU_TYPES`
> * `ForgeRegistries.CONTAINERS` -> `ForgeRegistries.MENU_TYPES`
> * `ForgeRegistries.ENTITIES` -> `ForgeRegistries.ENTITY_TYPES`
> * `ForgeRegistries.BLOCK_ENTITIES` -> `ForgeRegistries.BLOCK_ENTITY_TYPES`
> * `ForgeRegistries.PROFESSIONS` -> `ForgeRegistries.VILLAGER_PROFESSIONS`
> 
> `ForgeRegistries.PAINTING_TYPES` is weird because the Mojang classname is "Motive", not sure if we should make that name worse.
> 
> `ForgeRegistries.DATA_SERIALIZERS` is quite generic, I think it would be best to rename it to start with `ENTITY_`. Same thing for `ForgeRegistries.LOOT_MODIFIER_SERIALIZERS`, should this be prefixed with `GLOBAL_` to match the classname?

agree with everything except the block entity rename, it looks good as it is. and also the menu. CONTAINERS simply should be MENUS. keeping it simple :)

**Comment by josephtarbit** [comment]
> Some ForgeRegistries registries and keys are inconsistent relative to the classname / Key, so I think all of these changes are reasonable:
> 
> * `ForgeRegistries.Keys.CONTAINER_TYPES` -> `ForgeRegistries.Keys.MENU_TYPES`
> * `ForgeRegistries.CONTAINERS` -> `ForgeRegistries.MENU_TYPES`
> * `ForgeRegistries.ENTITIES` -> `ForgeRegistries.ENTITY_TYPES`
> * `ForgeRegistries.BLOCK_ENTITIES` -> `ForgeRegistries.BLOCK_ENTITY_TYPES`
> * `ForgeRegistries.PROFESSIONS` -> `ForgeRegistries.VILLAGER_PROFESSIONS`
> 
> `ForgeRegistries.PAINTING_TYPES` is weird because the Mojang classname is "Motive", not sure if we should make that name worse.
> 
> `ForgeRegistries.DATA_SERIALIZERS` is quite generic, I think it would be best to rename it to start with `ENTITY_`. Same thing for `ForgeRegistries.LOOT_MODIFIER_SERIALIZERS`, should this be prefixed with `GLOBAL_` to match the classname?

I think `PAINTING_TYPES` should be renamed to `PAINTING_VARIANTS` as that's the new name in the 1.19 snapshots.

**Comment by ghost** [comment]
> > Some ForgeRegistries registries and keys are inconsistent relative to the classname / Key, so I think all of these changes are reasonable:
> > 
> > * `ForgeRegistries.Keys.CONTAINER_TYPES` -> `ForgeRegistries.Keys.MENU_TYPES`
> > * `ForgeRegistries.CONTAINERS` -> `ForgeRegistries.MENU_TYPES`
> > * `ForgeRegistries.ENTITIES` -> `ForgeRegistries.ENTITY_TYPES`
> > * `ForgeRegistries.BLOCK_ENTITIES` -> `ForgeRegistries.BLOCK_ENTITY_TYPES`
> > * `ForgeRegistries.PROFESSIONS` -> `ForgeRegistries.VILLAGER_PROFESSIONS`
> > 
> > `ForgeRegistries.PAINTING_TYPES` is weird because the Mojang classname is "Motive", not sure if we should make that name worse.
> > `ForgeRegistries.DATA_SERIALIZERS` is quite generic, I think it would be best to rename it to start with `ENTITY_`. Same thing for `ForgeRegistries.LOOT_MODIFIER_SERIALIZERS`, should this be prefixed with `GLOBAL_` to match the classname?
> 
> I think `PAINTING_TYPES` should be renamed to `PAINTING_VARIANTS` as that's the new name in the 1.19 snapshots.

Forge should refer to them as MOTIVES

**Comment by sciwhiz12** [comment]
RE: `PAINTING_TYPES`, we'll see what final name will be used when Forge starts updating for 1.19. We will most likely follow the name of the vanilla Registry, so it all matches as one would expect.

**Comment by Luis-St** [comment]
> @Luis-st: Instead of renaming the existing getters, it may perhaps be better to override `#getEntity` and narrow down the return type instead to e.g. `ItemEntity` and `LivingEntity` (and perhaps `PlayerEvent`) using the covariance of method return types. What do you think?

looks like a better solution, the old getter's could then be marked as `Deprecated` and removed in a futher version, 
maybe 1.20 or 1.21?

**Comment by Luis-St** [comment]
`SidedInvWrapper` -> `WorldlyInvWrapper`

**Comment by AterAnimAvis** [comment]
As mentioned by Sci in MMD, `Gui` -> `Screen` in *at least* the following places (likely to be several more scattered arround the code base):

`ConfigGuiFactory` -> `ConfigScreenFactory`
`ConfigGuiFactory#getGuiFactoryFor` -> `#getScreenFactoryFor`
`NetworkHooks#openGui` -> `#openScreen`

**Comment by josephtarbit** [comment]
`RegisterCommandsEvent#getEnvironment` -> `#getCommandSelection`

Reason: The method returns an instance of `CommandSelection`.

**Comment by alcatrazEscapee** [comment]
`TagsUpdatedEvent.registries` -> `registryAccess`
`TagsUpdatedEvent#getTagManager` -> `#getRegistryAccess`

Reason: `TagsUpdatedEvent` has:
- a field of type `RegistryAccess`
- which is named `registries` (vanilla doesn't refer to it as registries, rather in most if not all fields it's called `registryAccess`)
- with a accessor called `getTagManager` (which is just completely wrong)
- with a javadoc saying "Returns the dynamic registries" (which is a pre-1.17 name)

consistency please.

**Comment by KnightMiner** [comment]
Not sure if this fits here, but its almost a method rename:

`HumanoidArmorLayer#getArmorResource` -> make static instead of non-static. Nothing in the method depends on the parent model, and it would be useful to call in an armor renderer for an entity that does not use the normal humanoid renderer.

**Comment by IchHabeHunger54** [comment]
Package `net.minecraftforge.fml.javafmlmod` -> `net.minecraftforge.fml.fmljavamod`

Class `FMLModContainer` -> `FMLJavaModContainer`

Reason: Consistency with `FMLJavaModLoadingContext` and `FMLJavaModLanguageProvider`, which are located in this package as well

**Comment by Random832** [comment]
In ForgeRegistries and ForgeRegistries.Keys

Field `CONTAINERS` -> `MENUS`
Field `CONTAINER_TYPES` -> `MENU_TYPES` [but see below]
Reason: consistency with vanilla/mojmaps

There are also wider issues to consider in the naming conventions of forge registries, which should possibly be examined at some point

1. Use of "TYPE" or not is inconsistent between static registry constants and keys. These should be the same in all cases.
2. Use of "TYPE" is inconsistent with vanilla. It's possible that Vanilla makes the wrong call here [the vanilla registry is MENU but the registered class is MenuType], but deviations from vanilla registry names should be carefully considered and have a guiding principle behind them
3. Most (all?) forge registry names are plural whereas the vanilla one is singular. Changing this would cause a lot of unnecessary churn for mod maintainers, but it is a deviation from vanilla.

**Comment by amadornes** [comment]
I've been going through all the client code as part of my audit and here are the names I would suggest changing.  
Sorry for the massive list, it was a big audit 😅

Per-rename notes are only provided if considered necessary, since in a lot of cases there is not much to explain that isn't already signified by the category title. I'd be more than happy to provide an explanation for anything else if necessary, though. Just let me know.

### Match Mojang's own mappings and/or naming scheme:
 - `EffectRenderer` -> `MobEffectRenderer`
 - `ItemModelMesherForge` -> `ForgeItemModelShaper` (already suggested some time ago, but not implemented)
 - `ContainerScreenEvent`
    - `DrawBackground` -> `RenderBackground`
    - `DrawForeground` -> `RenderForeground`
 - `DrawSelectionEvent` -> `RenderHighlightEvent`
    - `HighlightBlock` -> `Block` (simplify, since the parent event would be renamed)
    - `HighlightEntity` -> `Entity` (simplify, since the parent event would be renamed)
 - `ParticleFactoryRegisterEvent` -> `RegisterParticleProvidersEvent`
 - `RenderBlockOverlayEvent` -> `RenderBlockScreenEffectEvent`
 - `RenderGameOverlayEvent`
    - `BossInfo` -> `BossEventProgress` (this is used for raids as well now)
 - `RenderNameplateEvent` -> `RenderNameTagEvent`
 - `ForgeIngameGui` -> `ForgeGui`
 - `GuiUtils` -> `ScreenUtils`
    - `#drawContinuousTexturedBox` -> `blitWithBorder`
    - `#drawInscribedRect` -> `blitInscribed`
 - `IIngameOverlay` -> `IGuiOverlay`
 - `IModelConfiguration` (this gets renamed below, here are only the changes to match mojmap)
    - `#isTexturePresent` -> `hasMaterial`
    - `#resolveTexture` -> `getMaterial`
    - `#isShadedInGui` -> `isGui3d`
    - `#useSmoothLighting` -> `hasAmbientOcclusion`
    - `#getCameraTransforms` -> `getTransforms`
 - `IModelGeometry` -> `IUnbakedGeometry` (IModel -> UnbakedModel)
    - `#getParts` -> `getComponents` ("part" to "component" transition [*1])
    - `#getPart` -> `getComponent` ("part" to "component" transition [*1])
 - `ISimpleModelGeometry` -> `ISimpleUnbakedGeometry`
 - `IModelGeometryPart` -> `IUnbakedGeometryComponent`
 - `IMultipartModelGeometry` -> `IUnbakedCompositeGeometry` ("part" to "component" transition [*1])
 - `KeyBindingMap` -> `KeyMappingLookup` (match mojmap but avoid having a mapping map, simplify member names)
    - `#addKey` -> `add`
    - `#clearMap` -> `clear`
    - `#getBinding` -> `find`
    - `#lookupActive` -> `findActive`
    - `#lookupAll` -> `findAll`
    - `#removeKey` -> `remove`

### More accurate and/or descriptive names:
 - `IBlockRenderProperties` -> `IBlockClientExtensions` (nothing here is strictly rendering-related)
 - `IItemRenderProperties` -> `IItemClientExtensions` (not everything is strictly rendering-related, and match previous)
    - `#getArmorModel` -> `getHumanoidArmorModel`
    - `#getBaseArmorModel` -> `getGenericArmorModel`
    - `#getItemStackRenderer` -> `getCustomRenderer` (match isCustomRenderer in BakedModel)
 - `ClientPlayerNetworkEvent`
    - `LoggedInEvent` -> `LoggingIn` (remove redundant "event", and change to continuous tense since it fires in the middle of the login process when the player isn't even in the level yet)
    - `LoggedOutEvent` -> `LoggingOut` (remove redundant "event" and change to continuous tense since it fires before the player has completely logged out)
    - `RespawnEvent` -> `Clone` (match PlayerEvent since this also gets fired when changing dimensions)
 - `ColorHandlerEvent` -> `RegisterColorHandlersEvent` (registration events should be prefixed by "register")
 - `EntityViewRenderEvent` -> `ViewportEvent`
    - `RenderFogEvent` -> `RenderFog` (remove redundant "event")
    - `FogColors` -> `ComputeFogColor` (event names should reflect an action or hook point, not a property or state)
    - `CameraSetup` -> `ComputeCameraAngles` (ditto)
    - `FieldOfView` -> `ComputeFOV` (ditto)
 - `FOVModifierEvent` -> `ComputeFOVModifierEvent` (ditto)
 - `InputEvent`
    - `RawMouseEvent` -> `MouseButtonPre` (these should be exempt of the previous point since they carry an "action" inside)
    - `MouseInputEvent` -> `MouseButtonPost` (ditto, both of these are to be replaced with a proper bi-phase hierarchy)
    - `MouseScrollEvent` -> `MouseScrolling`
    - `KeyInputEvent` -> `Key` (remove redundant "event" and normalize names)
    - `ClickInputEvent` -> `InteractionKeyMappingTriggered` (this is hard to name, but it's clearer imo)
 - `ModelBakeEvent` -> `ModelBakingCompletedEvent` (more specific about when this gets fired)
 - `RenderGameOverlayEvent`
    - `Text` -> `DebugText`
 - `ScreenEvent`
    - `InitScreenEvent` -> `Init` (remove redundant "screen event")
    - `DrawScreenEvent` -> `Render`  (remove redundant "screen event", match mojmap)
    - `BackgroundDrawnEvent` -> `BackgroundRendered` (remove redundant "event", match mojmap)
    - `MouseInputEvent` -> `MouseInput` (remove redundant "event")
    - `MouseClickedEvent` -> `MousePressed` (remove redundant "event", more descriptive name)
    - `MouseReleasedEvent` -> `MouseReleased` (remove redundant "event")
    - `MouseDragEvent` -> `MouseDrag` (remove redundant "event")
    - `MouseScrollEvent` -> `MouseScroll` (remove redundant "event")
    - `KeyboardKeyEvent` -> `KeyInput` (more descriptive name)
    - `KeyboardKeyPressedEvent` -> `KeyPressed` (remove redundant "keyboard" and "event")
    - `KeyboardKeyReleasedEvent` -> `KeyReleased` (remove redundant "keyboard" and "event")
    - `KeyboardCharTypedEvent` -> `CharacterTyped` (remove redundant "event")
 - `ScreenOpenEvent` -> `ScreenOpeningEvent` (change to continuous tense since it's an ongoing action that may be canceled)
 - `ForgeIngameGui`
    - `#<something>_ELEMENT` -> `<something>_OVERLAY` (these are all overlays, they should be named as such)
    - `#BOSS_HEALTH_ELEMENT` -> `BOSS_EVENT_PROGRESS_OVERLAY` (match mojmap, this is used for raids as well now)
    - `#HUD_TEXT_ELEMENT` -> `DEBUG_TEXT_OVERLAY`
    - `#left_height` -> `leftHeight` (fix casing)
    - `#right_height` -> `rightHeight` (fix casing)
 - `NotificationModUpdateScreen` -> `TitleScreenModUpdateIndicator` (not a standalone screen)
 - `IModelConfiguration` -> `IGeometryBakingContext` (discussed with gigaherz, this feels more appropriate name)
    - `#getPartVisibility` -> `getComponentVisibility` ("part" to "component" transition [*1])
 - `StandaloneModelConfiguration` -> `StandaloneGeometryBakingContext`
 - `BlockModelConfiguration` -> `BlockModelGeometryBakingContext`
    - `VisibilityData` -> `ComponentVisibilityData` (more descriptive, "part" to "component" transition [*1])
 - `IGeneratedBlockstate` -> `IGeneratedBlockState` (fix casing)
 - `OBJModel` -> `ObjModel` (match vanilla casing NBT -> Nbt)
 - `OBJLoader` -> `ObjLoader`
 - `MaterialLibrary` -> `ObjMaterialLibrary` (specify it's part of the OBJ model system)
 - `LineReader` -> `ObjTokenizer` (specify it's part of the OBJ model system, more descriptive name as it doesn't just read lines)
 - `OBJLoaderBuilder` -> `ObjModelBuilder` (it builds models, not loaders)
 - `BakedRenderable` -> `BakedModelRenderable` (wraps BakedModel into an IRenderable)
 - `SimpleRenderable` -> `CompositeRenderable` ("part" to "component" transition [*1])
 - `UnitSprite` -> `UnitTextureAtlasSprite`
 - `GuiUtils` (this gets renamed above)
    - `#colorCodes` -> `TEXT_COLOR_CODES`
    - `#getColorCode` -> `getColorFromFormattingCharacter`

\*1. The reason for the "part" to "component" transition is that vanilla has multipart blockstates, and it would be best to differentiate all renamed elements since they have nothing to do with it. A "composite model" is a model composed of multiple components/parts, so I feel it's an appropriate naming scheme to use.

**Comment by IchHabeHunger54** [comment]
> `RawMouseEvent` -> `MouseButton`

Should be named `MouseButtonPre` or something similar instead, as I feel that having a Post event, but not a Pre event is weird.

> `KeyboardKeyEvent` -> `KeyboardInput`

I'd go with `KeyInput` instead, because that seems more consistent with `KeyPressed` and `KeyReleased`.

**Comment by amadornes** [comment]
So... the reasoning for the mouse button one is that, if my changes go through, I'll be nuking the other event since it has no reason to exist. It's a leftover from way back, and `RawMouseEvent` was added on later because the original event didn't allow for cancellation and they didn't want to change it mid-version. There's no practical use case for the post event I can think of outside of perhaps checking if a click resulted in some change, but seeing as that currently doesn't apply to any of the other events (scroll and keyboard), it doesn't make a whole lot of sense to keep it around.

If it's decided that it should be kept, then yes - `MouseButtonPre` would be a much better name. Better yet, it should be changed to a hierarchy with `MouseButton` containing `Pre` and `Post` like all other bi-phase events.

You're right on `KeyInput`, though. I'll update my original comment with that.

**Comment by marchermans** [comment]
@amadornes I am okey with all of them, allthough I find `ClientPlayerEvent` a bit smelly...

In principle you are right, they are about more then just networking, but realistically they indicate that a packet to perform something on the client side has arrived.

So I am pretty happy with the name that outer class has, but I am also not against renaming it. Just not to ClientPlayerEvent.

Especially since the concept of a player event already exists and that is tied to an interaction which gets triggered by a player entity, not by a packet arriving.

TLDR: I would not veto the names, but if you could come up with a better one for `ClientPlayerEvent` that would be appreciated.

**Comment by Lanse505** [comment]
Just a suggestion but what about:
``ClientPlayerTriggerEvent``

This would mark the event as a "TriggerEvent" indicating its an event fired when something triggers in regard to the Player on the Client-side. In this case on Login, Logout and "clone" (respawn).

This also opens up for Server-sided trigger events such as ServerWorldTriggerEvent for future triggers in regard to the world for example saving and such?

I'm not super happy with ``ClientPlayerEvent`` either but I cant realistically think of a better name that doesn't just become convoluted either. This was the best alternative I could come up with and it isnt great either since it introduces a the concept of a "event" fired by a piece of "trigger" code, which Im not sure is a concept we realistically can justify being a definition, since what realistically constitutes a piece of "trigger" code?

Anyway, maybe something to think about?

**Comment by amadornes** [comment]
As per everyone's feedback, I've rolled back the change to `ClientPlayerNetworkEvent`. I feel if we want to keep the notion that it's triggered by a network packet in place, that's the best name to go with.

I've also changed `RawMouseButton` from `MouseButton` to `MouseButtonPre` after discussing it with Orion earlier and deciding the old event should stick around. If my audit + recommended changes get approved, I'll change this into a proper hierarchy with `MouseButton` containing both pre and post events.

**Comment by Random832** [comment]
Class: `IForgeItem` and `IForgeItemStack`
- `getContainerItem` -> `getCraftingRemainingItem`
- `hasContainerItem` -> `hasCraftingRemainingItem`

**Comment by sciwhiz12** [comment]
These renames were performed across #8786 (the client refactors PR) and #8840 (the general renames PR).

@amadornes has provided a guide for automatically remapping your environment using a provided TSRGv2 file to use the new names: https://gist.github.com/amadornes/cead90457e766f6d4294cb6b812f91dc. 

The guide already includes the TSRGv2 files for both the client refactors and general renames, so users running this on `41.0.64` and below will need to update to `41.0.94` or newer, as indicated in the guide.
